### PR TITLE
Use dict_factory in store for better readability

### PR DESCRIPTION
**Pull request by Bouni**
I always struggled with al those index numbers scatterd all around the code base and for a long time wanted to improve that.

So I started with the store and introduced a [row_factory](https://docs.python.org/3/library/sqlite3.html#sqlite3.Connection.row_factory) that returns a dict instead of a list which improves readability throughout the code base a lot.

For example

`if self.hide_bom_parts and part[6]:`

becomes

`if self.hide_bom_parts and part["exclude_from_bom"]:`

### CODE REVIEW

File: store.py [hunk 1]
```diff
@@ -85,13 +86,11 @@ def read_all(self):
         """Read all parts from the database."""
```
Comment (chmorgan) [inline store.py:86]: should we annotate this with a '-> dict' hint?

File: store.py [hunk 2]
```diff
@@ -183,57 +184,77 @@ def set_lcsc(self, ref, value):
...
+            db_part = self.get_part(board_part["reference"])
             # if part is not in the database yet, create it
-            if not dbpart:
+            if not db_part:
                 self.logger.debug(
                     "Part %s does not exist in the database and will be created from the board.",
-                    part[0],
+                    board_part["reference"],
                 )
+<<<<<<< HEAD
```
Comment (chmorgan) [inline store.py]: looks like a merge conflict snuck in here

File: store.py [hunk 3]
```diff
@@ -122,17 +122,17 @@ def create_part(self, part):
...
-                    "UPDATE part_info set value = ?, footprint = ?, lcsc = ?, exclude_from_bom = ?, exclude_from_pos = ? WHERE reference = ?",
-                    part[1:] + part[0:1],
-                )
-            else:
-                cur.execute(
-                    "UPDATE part_info set value = ?, footprint = ?, exclude_from_bom = ?, exclude_from_pos = ? WHERE reference = ?",
-                    part[1:] + part[0:1],
-                )
-
+            cur.execute(
```
Comment (chmorgan) [inline store.py:116]: this is WAY simpler than the original version.

File: mainwindow.py [hunk 4]
```diff
@@ -779,7 +779,7 @@ def toggle_bom_pos(self, *_):
             fp = board.FindFootprintByReference(ref)
             bom = toggle_exclude_from_bom(fp)
             pos = toggle_exclude_from_pos(fp)
-            self.store.set_bom(ref, bom)
+            self.store.set_bom(ref, int(bom))
```
Comment (chmorgan) [inline mainwindow.py:782]: what was up with the necessity to cast here but not previously? (No mention in the commit message of the reason why)

File: store.py [hunk 5]
```diff
@@ -178,13 +178,6 @@ def update_from_board(self):
                     "Part %s does not exist in the database and will be created from the board.",
                     board_part["reference"],
                 )
-<<<<<<< HEAD
```
Comment (chmorgan) [inline store.py]: you could do a rebase and fixup this commit into the earlier one where the merge conflict was introduced, if you are comfortable with using git like that.

**Review by chmorgan — COMMENTED** [review]
A handful of minor comments, this is a huge cleanup. I like the idea of using the columns directly from the database to generate the dict, that way if the order does ever change none of the code will have to.

File: store.py (hunk 1 above)
Comment (Bouni) [reply store.py:86]: Absolutely, will add that in a minute

File: store.py (hunk 3 above)
Comment (Bouni) [reply store.py:116]: Thats the point of this PR, I should have used a dict factory from the beginning, but wasn't aware that this is even possible ...

File: mainwindow.py (hunk 4 above)
Comment (Bouni) [reply mainwindow.py:782]: I don't remember but I guess that sqlite doesn't like to get a "1" for a boolean value. Previously the casting was done in th e set_bom function, in other places it was done like this. So I decided to make the cast always in the same place.

**Review by chmorgan — APPROVED** [review]

**Comment by Bouni** [comment]
Thanks @chmorgan for the review 👍🏽

**Pull request merged**